rmw_opensplice_cpp/src/functions.cpp

```diff
@@ -66,11 +68,10 @@ rmw_init()
 {
   DDS::DomainParticipantFactory_var dp_factory = \
     DDS::DomainParticipantFactory::get_instance();
-  if (!dp_factory)
-  {
+  if (!dp_factory) {
```

Comment:
The point that we didn't come to consensus on was the ability to put the `{` on a new line for `if`, `for`, and `while` if the parameters are wrapped for line length. For example:

```c++
if (true) {
  ...
}

if (
  SomeType a_really_long_variable_name_that_must_wrap_a,
  SomeOtherType a_really_long_variable_name_that_must_wrap_b)
{
  ...
}

if (
  SomeType a_really_long_variable_name_that_must_wrap_a,
  SomeOtherType a_really_long_variable_name_that_must_wrap_b
) {
  ...
}
```

My point was that the third instance would be good if you could have a trailing `,` on the last parameter, which would minimize the diff. But since that isn't the case putting the closing `)` on the same line as the last parameter doesn't affect the diff-ability. Also the second if statement aligns better with the multi-line function call style:

```c++
int ret = some_function(
  SomeType a_really_long_variable_name_that_must_wrap_a,
  SomeOtherType a_really_long_variable_name_that_must_wrap_b);
```

Honestly, I got in the habit of always putting the `{` on a new line in all cases, because that's an easy rule to follow, but I don't really care about that. However, I do think that the second `if` statement above is preferable to the third one, but again I'm willing to go with the flow if others disagree.

One option is that we enforce `{` on a new line for the `namespace`, `class`, and `struct` and just don't enforce style on the others, leaving the style up to the developer, since the best style to use seems to depend on the scenario.